apps/ledger-live-desktop/src/renderer/components/TransactionConfirm/index.tsx

```diff
@@ -137,7 +168,88 @@ const TransactionConfirm = ({ t, device, account, parentAccount, transaction, st
   });
   const key = ("mode" in transaction && transaction.mode) || "send";
   const recipientWording = t(`TransactionConfirm.recipientWording.${key}`);
-  return (
+
+  console.log(fields);
+
+  const typeTransaction: string =
+    (
+      fields.find(
+        (field: { label: string }) => field.label && field.label === "Type",
+      ) as DeviceTransactionField & { value: string }
+    )?.value || "";
+
+  const amountTransaction: string =
+    (
+      fields.find(
+        (field: { label: string }) => field.label && field.label === "Amount",
+      ) as DeviceTransactionField & { value: string }
+    )?.value || "";
+  return typeTransaction === "Approve" ? (
+    <Container style={{ paddingBottom: 0 }}>
+      <Container paddingX={26}>
+        <DeviceBlocker />
+        <Animation animation={getDeviceAnimation(device.modelId, type, "verify")} />
+        <Text ff={"Inter|Medium"} textAlign={"center"} fontSize={22} marginBottom={12}>
+          {t("approve.description")}
+        </Text>
+        <Alert type="primary" mb={26}>
+          <Trans
+            i18nKey={
+              amountTransaction === "Unlimited MATIC" ? "approve.unlimited" : "approve.limited"
+            }
+            values={{
+              recipientWording,
+            }}
+          />
+        </Alert>
+        <Box
+          style={{
+            width: "100%",
+          }}
+          mb={20}
+        >
+          {fields.map((field, i) => {
+            const MaybeComponent = fieldComponents[field.type];
+            if (!MaybeComponent) {
+              console.log(
+                `TransactionConfirm field ${field.type} is not implemented! add a generic implementation in components/TransactionConfirm.js or inside families/*/TransactionConfirmFields.js`,
+              );
+              return null;
+            }
+            return (
+              <MaybeComponent
+                key={i}
+                field={field}
+                account={account}
+                parentAccount={parentAccount}
+                transaction={transaction}
+                status={status}
+              />
+            );
+          })}
+        </Box>
+      </Container>
+      {Footer ? (
+        <>
+          <HorizontalSeparator />
+          <Footer transaction={transaction} />
+        </>
+      ) : manifestId && Object.keys(termsOfUse).includes(manifestId) ? (
+        <>
+          <HorizontalSeparator />
+          <Text marginTop={30}>
+            <Trans i18nKey="approve.warning" />{" "}
+            <Link
+              onClick={() => openURL(termsOfUse[manifestId as keyof typeof termsOfUse])}
```

Comment:
I'm not a fan of using the `as` keyword like that, I think you can avoid the issue by adding an `as const` on the `termsOfUse` object
```suggestion
              onClick={() => openURL(termsOfUse[manifestId])}
```

Reply:
[Let's merge this conv](https://github.com/LedgerHQ/ledger-live/pull/6174#discussion_r1490692229)